File: internal/httpapi/handlers/login_log.go
package handlers

import "github.com/gin-gonic/gin"

func (h *Handler) RegisterLoginLog(r gin.IRouter) {
	r.GET("/api/admin/login-logs", h.loginLogs)
	r.DELETE("/api/admin/login-logs", h.clearLoginLogs)
}
func (h *Handler) loginLogs(c *gin.Context) {
	if h.app().Users == nil {
		writeNotConfigured(c)
		return
	}
	data, total, err := h.app().Users.GetLoginLogs(c.Request.Context(), queryInt(c, "limit", 100), queryInt(c, "offset", 0))
	if err != nil {
		writeError(c, err)
		return
	}
	writeOK(c, gin.H{"items": data, "total": total})
}
func (h *Handler) clearLoginLogs(c *gin.Context) {
	if h.app().Users == nil {
		writeNotConfigured(c)
		return
	}
	if err := h.app().Users.ClearLoginLogs(c.Request.Context()); err != nil {
		writeError(c, err)
		return
	}
	writeOK(c, nil)
}

File: web/src/api/login-log.ts
import type { ApiEnvelope, ApiRequestConfig } from './types'
import client from './client'

export function getLoginLogs<T = any>(limit = 100, offset = 0, config?: ApiRequestConfig) {
  return client.get<ApiEnvelope<T>>('/api/admin/login-logs', {
    ...config,
    params: { ...config?.params, limit, offset },
  })
}

export function clearLoginLogs<T = any>(config?: ApiRequestConfig) {
  return client.delete<ApiEnvelope<T>>('/api/admin/login-logs', config)
}
